src/Worker.Extensions.DurableTask/TaskEntityDispatcher.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.DurableTask;
+
+/// <summary>
+/// Represents a task entity dispatch invocation.
+/// </summary>
+public sealed class TaskEntityDispatcher
+{
+    private readonly string request;
+    private readonly IServiceProvider services;
+
+    internal TaskEntityDispatcher(string request, IServiceProvider services)
+    {
+        this.request = request;
+        this.services = services;
+    }
+
+    internal string Result { get; private set; } = string.Empty;
+
+    /// <summary>
+    /// Dispatches this entity trigger to the provided <see cref="ITaskEntity"/>.
+    /// </summary>
+    /// <param name="entity">The task entity to dispatch to.</param>
+    /// <returns>A task that completes when the dispatch has finished.</returns>
+    public async Task DispatchAsync(ITaskEntity entity)
+    {
+        if (entity == null)
+        {
+            throw ArgumentNullException(nameof(entity));
+        }
+
+        this.Result = await GrpcEntityRunner.LoadAndRunAsync(this.request, entity);
+    }
+
+    /// <summary>
+    /// <para>Dispatches the entity trigger to an instance of the provided <typeparamref name="T"/>.</para>
+    /// <para>
+    /// If <typeparamref name="T"/> is <see cref="ITaskEntity"/>, it will be activated from <see cref="IServiceProvider"/>
+    /// and then be ran.
+    /// </para>
+    /// <para>
+    /// If <typeparamref name="T"/> is not <see cref="ITaskEntity"/>, it is assumed the type represents the entity state
+    /// and it will be deserialized and dispatched directly to the state.
+    /// </para>
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <returns></returns>
+    public Task DispatchAsync<T>()
+    {
+        if (typeof(ITaskEntity).IsAssignableFrom(typeof(T)))
+        {
+            ITaskEntity entity = (ITaskEntity)ActivatorUtilities.GetServiceOrCreateInstance<T>(this.services);
+            return this.DispatchAsync(entity);
+        }
+
+        return this.DispatchAsync(new StateEntity<T>());
```

Comment:
I'm definitely a tad confused about the different cases for `T` here. 

It appears that in the `if` case `<T>` corresponds to a user-level Entity class declaration, like the standard `CounterEntity` class for example. Do I have that right?

I'm not sure I understand what the else-case corresponds to though. I'd appreciate an explanation there :) . Thanks!

Reply:
Thanks, this makes sense.  think it would be good to add a version of this as inlined comment in the implementation.

Reply:
For comments, I would prefer we write msdocs and then add a link to that doc in the comments.

Reply:
+1 for deferring further revisions of the XML comments until after we have updated the msdocs. I think this is something we probably want to do for the entire feature branch.

@davidmrdavid , there is an example for a simple counter  here: https://github.com/Azure/azure-functions-durable-extension/pull/2584